app/lib/audit/backend.dart

```diff
@@ -93,14 +100,34 @@ class AuditBackend {
     );
   }
 
+  @visibleForTesting
+  String nextTimestamp(DateTime last, DateTime next) {
+    final nextDayStart =
+        DateTime.utc(next.year, next.month, next.day).add(Duration(days: 1));
+    return nextDayStart.isBefore(last) && nextDayStart.isAfter(next)
+        ? nextDayStart.toIso8601String().split('T').first
+        : next.toIso8601String();
+  }
+
   /// Parses the `before` query parameter and returns the parsed timestamp.
   ///
   /// Returns a timestamp slightly into the future if the parameter is missing.
   /// Returns null if the query parameter is invalid.
   DateTime? parseBeforeQueryParameter(String? param) {
+    final now = DateTime.now().toUtc();
     if (param == null) {
-      return DateTime.now().add(const Duration(minutes: 5));
+      return now.add(const Duration(minutes: 5));
+    }
+    if (param.length == 10) {
+      final m = _shortBeforeFormat.matchAsPrefix(param);
+      if (m != null) {
+        final parsed = DateTime.utc(int.parse(m.group(1)!),
+            int.parse(m.group(2)!), int.parse(m.group(3)!));
+        if (parsed.year > 2000 && parsed.year <= now.year) {
+          return parsed;
+        }
```

Comment:
Isn't it better to throw a `ResponseException` complaining that years before 2000 and after current year, is not allowed.